src/workflows/airqo_etl_utils/airqo_utils.py

```diff
@@ -370,15 +373,15 @@ def extract_devices_data(
                         )
                     )
 
-                data["device_number"] = device_number
-                data["device_id"] = device.get("device_id")
-                data["site_id"] = device.get("site_id")
+                meta_data = data.attrs.get("meta_data", {})
 
-                if device_category == DeviceCategory.BAM:
-                    data["latitude"] = meta_data.get("latitude", None)
-                    data["longitude"] = meta_data.get("longitude", None)
+                data["device_number"] = device.get("device_number", None)
+                data["device_id"] = device.get("device_id", None)
```

Comment:
Is `device.get("device_number", None)` necessary? At this point you already have device_number in a variable defined at the beginning of this loop.